phpBB/phpbb/storage/adapter/local.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\storage\adapter;
+
+use phpbb\storage\exception\exception;
+use phpbb\filesystem\exception\filesystem_exception;
+
+class local implements adapter_interface
+{
+	/**
+	 * Filesystem component
+	 *
+	 * @var \phpbb\filesystem\filesystem
+	 */
+	protected $filesystem;
+
+	/** @var string path */
+	protected $root_path;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct(\phpbb\config\config $config, \phpbb\filesystem\filesystem $filesystem, $phpbb_root_path, $path_key)
+	{
+		$this->filesystem = $filesystem;
+		$this->root_path = $phpbb_root_path.$config[$path_key];
+
+		if (substr($this->root_path, -1, 1) != DIRECTORY_SEPARATOR)
+		{
+			$this->root_path = $this->root_path.DIRECTORY_SEPARATOR;
+		}
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function put_contents($path, $content)
+	{
+		if ($this->exists($path))
+		{
+			throw new exception('', $path); // FILE_EXISTS
+		}
+
+		try
+		{
+			$this->filesystem->dump_file($this->root_path.$path, $content);
+		}
+		catch (filesystem_exception $e)
+		{
+			throw new exception('', $path, array(), $e); // CANNOT_DUMP_FILE
+		}
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function get_contents($path)
+	{
+		if (!$this->exists($path))
+		{
+			throw new exception('', $path); // FILE_DONT_EXIST
+		}
+
+		if (($content = @file_get_contents($this->root_path.$path)) === false)
```

Comment:
we don't really uses this kind of notation here, usually we prefer to split it into 2 lines